runtime/lua/vim/lsp/buf.lua

```diff
@@ -901,11 +991,16 @@ local function hierarchy(method)
   local bufnr = api.nvim_get_current_buf()
   local clients = lsp.get_clients({ bufnr = bufnr, method = prepare_method })
   if not next(clients) then
-    vim.notify(lsp._unsupported_method(method), vim.log.levels.WARN)
+    local msg = lsp._unsupported_method(method)
+    ---@type lsp.ResponseError
+    local error = { code = lsp.protocol.ErrorCodes.RequestCancelled, message = msg }
+    invoke_callback(callback, { error })
+    vim.notify(msg, vim.log.levels.WARN)
```

Comment:
This is slightly inconsistent with line 711, where we call `vim.notify` but do not append to `errors`, though should be solved when we move all the `vim.notify` calls.

Reply:
Do you want me to remove the `vim.notify` calls in this PR or later in a follow-up?